### DISCUSSION TITLE
Challenges with using a LetsEncrypt SSL  certificate generated outside of Modoboa for SMTP and IMAP

### DISCUSSION THREAD

**Discussion by beitaworks** [modoboa/modoboa#3330] [General]
Hi, I installed a certificate and updated the /etc/postfix/main.cf file and when i run openssl s_client -connect govmail.gbb.com.ng:587 -starttls smtp i get the lets encrypt certificate. But when i run openssl s_client -servername mydomain.com -connect mydomain.com:imaps i get the self-signed certificate on my server. The email clients i am trying to configure see the self-signed certificate and wont connect.
I am trying to simulate a situation where i can install my own SAN certificate without using letsencrypt then update postfix/dovecot with the fullchain and private key.
I need help please.

**Comment by jonas-t-s** [comment]
Did you tell Postfix and Dovecot to use your new certificate? 
Because IMAP(s) is managed by Dovecot and not by Postfix...

**Reply by beitaworks** [reply]
Yes I did in /etc/dovecot/conf.d/ 10-ssl.conf for dovecot and in /etc/postfix/main.cf

**Reply by jonas-t-s** [reply]
how do your dovecot ssl_cert, ssl_key and ssl lines look?

**Reply by beitaworks** [reply]
/etc/letsencrypt/live/mydomain/fullchain.pm

/etc/letsencrypt/live/mydomain/privkey.pem 

When I run on the server
openssl s_client -servername mydomain.com -connect mydomain.com:imaps
openssl s_client -connect mysomain.com:993 -crlf
openssl s_client -connect mydomain.com:587 -starttls smtp
I get the correct certificate however, when I try to setup any desktop or mobile client, they don't trust the certificate and i later found out that they were getting the self-signed certificate.

**Reply by beitaworks** [reply]
Dovecot

ssl = yes
ssl_cert = /etc/letsencrypt/live/mydomain.com/fullchain.pem
ssl_key = /etc/letsencrypt/live/mydomain.com/privkey.pem
!include_try /etc/dovecot/conf.d/10-ssl-keys.try

Postfix

smtp_use_tls = yes
smtp_tls_security_level = may
smtp_tls_session_cache_database = btree:$data_directory/smtp_scache
smtpd_use_tls = yes
smtpd_tls_auth_only = no
smtpd_tls_CApath = /etc/ssl/certs
smtpd_tls_key_file = /etc/letsencrypt/live/mydomain.com/privkey.pem
smtpd_tls_cert_file = /etc/letsencrypt/live/mydomain.com/fullchain.pem

I get this error when i output the log at the time a client wants to connect
tail -f /var/log/mail.log:
dovecot: imap-login: Disconnected: Connection closed: SSL_accept() failed: error:0A000416:SSL routines::sslv3 alert certificate unknown: SSL alert number 46 (no auth attempts in 0 secs): user=<>, rip=102.91.92.218, lip=10.0.146.11, TLS handshaking: SSL_accept() failed: error:0A000416:SSL routines::sslv3 alert certificate unknown: SSL alert number 46, session=<luKHlaMlRWBmW1za>
